apollo-runtime/src/main/java/com/apollostack/android/ApolloResponseBodyConverter.java

```diff
@@ -0,0 +1,32 @@
+package com.apollostack.android;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonReader;
+
+import java.io.IOException;
+
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+
+class ApolloResponseBodyConverter<T> implements Converter<ResponseBody, T> {
+  private final JsonAdapter<?> adapter;
+
+  ApolloResponseBodyConverter(JsonAdapter<?> adapter) {
+    this.adapter = adapter;
+  }
+
+  @Override public T convert(ResponseBody value) throws IOException {
+    try {
+      JsonReader reader = JsonReader.of(value.source());
+      reader.beginObject();
+      if (!reader.nextName().equals("data")) {
```

Comment:
Might be outside this PR's scope but worth mentioning: In some cases, the data field may [not](http://facebook.github.io/graphql/#sec-Data) exist.  Also, an errors array may be returned.